File: server/src/main/java/net/progruzovik/dissent/config/WebConfig.java
package net.progruzovik.dissent.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.converter.HttpMessageConverter;
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
import org.springframework.web.servlet.config.annotation.*;

import java.util.List;

@Configuration
@EnableWebMvc
public class WebConfig extends WebMvcConfigurerAdapter {

    private final ObjectMapper mapper;

    public WebConfig(ObjectMapper mapper) {
        this.mapper = mapper;
    }

    @Override
    public void addResourceHandlers(ResourceHandlerRegistry registry) {
        registry.addResourceHandler("/**").addResourceLocations("classpath:/static/");
    }

    @Override
    public void addViewControllers(ViewControllerRegistry registry) {
        registry.addViewController("/").setViewName("index.html");
    }

    @Override
    public void configureMessageConverters(List<HttpMessageConverter<?>> converters) {
        converters.add(new MappingJackson2HttpMessageConverter(mapper));
        super.configureMessageConverters(converters);
    }
}

File: server/src/main/kotlin/net/progruzovik/dissent/config/WebConfig.kt
package net.progruzovik.dissent.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.io.ClassPathResource
import org.springframework.http.MediaType
import org.springframework.web.reactive.function.server.HandlerFunction
import org.springframework.web.reactive.function.server.RequestPredicates.GET
import org.springframework.web.reactive.function.server.RouterFunction
import org.springframework.web.reactive.function.server.RouterFunctions.route
import org.springframework.web.reactive.function.server.ServerResponse
import org.springframework.web.reactive.function.server.ServerResponse.ok

@Configuration
class WebConfig {

    @Bean fun indexRouter(): RouterFunction<ServerResponse> {
        return route(
            GET(""),
            HandlerFunction { ok().contentType(MediaType.TEXT_HTML).syncBody(ClassPathResource("static/index.html")) }
        )
    }
}
